src/contexts/AuthContext.jsx

```diff
@@ -0,0 +1,66 @@
+import React, { createContext, useContext, useState, useEffect } from "react"
+import { createUserWithEmailAndPassword, signInWithEmailAndPassword, signOut, sendPasswordResetEmail, updateEmail, updatePassword, onAuthStateChanged } from "firebase/auth"
+import { auth } from "../firebase"
+
+const AuthContext = createContext()
+
+export function AuthProvider({ children }) {
+    const [currentUser, setCurrentUser] = useState({})
+    const [loading, setLoading] = useState(true)
+
+    function signup(email, password) {
+        console.log("here")
+        return createUserWithEmailAndPassword(auth, email, password)
+    }
+
+    function login(email, password) {
+        return signInWithEmailAndPassword(auth, email, password)
+    }
+
+    function logout() {
+        return signOut(auth)
+    }
+
+    function resetPassword(email) {
+        return sendPasswordResetEmail(auth, email)
+    }
+
+    function updateUserEmail(email) {
+        return updateEmail(auth.currentUser, email)
+    }
+
+    function updateUserPassword(password) {
+        return updatePassword(auth.currentUser, password)
+    }
+
+    useEffect(() => {
+        const unsubscribe = onAuthStateChanged(auth, (user) => {
+            setCurrentUser(user)
+            setLoading(false)
+        })
+
+        return () => {
+            unsubscribe()
+        }
+    }, [])
```

Comment:
I'm curious, do you not need to add auth or user into this array? Do you not want to rerun the effect when one of those values change?

Reply:
`onAuthStateChanged()` is an event listener so it will continue to check if the auth state changes. I only have to run that once in the beginning, but the auth changes as different users log in